icon_to_component.py

```diff
@@ -41,22 +43,27 @@ def create_component(name, svg):
 	svg = _fix_xmlns_xlink(svg, 'xmlns')
 	svg = _fix_xmlns_xlink(svg, 'xlink')
 	componentString = '''import React from "react";
-    import PropTypes from "prop-types";
-    import getIconStyle from "./iconStyle.js";
+import PropTypes, {{ any }} from "prop-types";
+import getIconStyle from "./iconStyle.js";
 
-    const {name} = props => (
-      <svg {{...getIconStyle(props.size)}} viewBox="0 0 16 16" {{...props}} onClick={{props.onClick}}>
+const {name} = (props) => {{
+  const {{ dimensions, viewBox, onClick, size, ...otherProps }} = props;
+  return (
+    <svg style={{{{ ...getIconStyle(size), ...dimensions }}}} viewBox={{(viewBox !== undefined) ? viewBox : "0 0 16 16"}} onClick={{onClick}} {{...otherProps}}>
```

Comment:
- `style={{{{ ...getIconStyle(size), ...dimensions }}}}` --> `style={{ ...getIconstyle(size), ...dimensions }}` 
   - first (and last) curly brace is to let React (JSX) know an expression is following
   - second (and second to last) curly brace indicate a short-hand object (needed to spread / merge the getIconStyle and dimensions objects)
   -  --> no need for 4 curly braces

- `viewBox={{(viewBox !== undefined) ? viewBox : "0 0 16 16"}}` --> `viewBox={viewBox  || "0 0 16 16"}`
   - your solution will compile to an invalid object because of the usage of two curly braces (see first remark for explanation)
   - The expression `viewBox  || "0 0 16 16"` is more concise way, if viewBox is `undefined / null / false` first part of the expression will evaluate to false  and the second part will be returned, if viewBox is defined, it will return it

- `onClick={{onClick}}` --> `onClick={onClick}`
   - Did you use Angular in the past? There an expression is indeed indicated with `{{<here comes my code>}}`, in React (JSX) however it's just `{<here comes my code>}`

- `{{...otherProps}}` --> `{...otherProps}`
   - Since we want to spread the props as attributes on svg tag, we don't need two curly braces (see previous remarks)

Reply:
- All double curly braces are because of python formatted string (curly braces are reserved characters).
- In this case, even when `dimensions` is not given as props, the style is just not spread by it (if that's what you meant).
- I used a ternary operator because of the case `<AssetPlaceholder viewBox="" />`. Then `viewBox || "0 0 16 16"` returns `"0 0 16 16"` and I want to override viewBox then. So it has to be compared with undefined I guess.